src/pages/NewsPage/index.tsx

```diff
@@ -1,117 +1,168 @@
+			className="w-36 md:h-14 md:w-48 rounded-[1.5rem] text-sm md:text-2xl bg-[#ca3735] hover:bg-[#ca37356e] text-center text-white hover:text-black hover:border-black px-4"
+			onClick={onClick}
+		>
 			<div className="button-text">{title}</div>
-			{thumbnailImage && (
-				<div className="image-container">
-					<img
-						className={imageClassName}
-						src={thumbnailImage}
-						alt={title} // Provide alt text for accessibility
-					/>
-				</div>
-			)}
 		</button>
 	);
 };
 
 interface NewsItemProps {
-	thumbnail: string;
 	title: string;
-	newsLink: string;
+	urlToImage: string;
+	url: string;
+	publishedAt: string;
 }
-const NewsItem: React.FC<NewsItemProps> = ({ thumbnail, title, newsLink }) => {
+
+const NewsItem: React.FC<NewsItemProps> = ({ title, urlToImage, url, publishedAt}) => {
+	const navigate = useNavigate(); // Initialize useNavigate hook
+
+	const openArticlePage = async (event: React.MouseEvent<HTMLAnchorElement>) => {
+		event.preventDefault();
+		// Fetch the news content from the provided URL
+		const response = await fetch(
+		'https://asia-southeast2-ntu-eee-dip-e028.cloudfunctions.net/dip-backend-functions/fetch_article?url=' + encodeURIComponent(url)
+		);
+		const newsContent = await response.text();
+		navigate(`/news/article?url=${encodeURIComponent(url)}&content=${encodeURIComponent(newsContent)}&title=${encodeURIComponent(title)}&urlToImage=${encodeURIComponent(urlToImage)}&publishedAt=${encodeURIComponent(publishedAt)}`); // Navigate to the article page with the URL parameter
+	};
+
 	return (
-		<div className="flex items-center justify-between p-4 border-solid border-red-200 hover:border-2 rounded-3xl px-4">
-			{/* Thumbnail */}
-			<div className="flex-shrink-0 mr-4 rounded-md">
+		<div className="flex flex-col md:flex-row item-center my-10 md:my-12 xl:px-20 xl:py-8 xl:my-8 border border-solid border-red-300 rounded-3xl">
+			{/* Image */}
+			<div className="flex">
 				<img
-					src={thumbnail}
+					src={urlToImage}
 					alt={title}
-					className="w-56 h-32 rounded-3xl mr-4"
+					className="h-40 w-52 mx-auto mt-4 xl:h-56 xl:w-96 md:h-40 md:w-60 md:my-4 md:mx-4 rounded-3xl xl:ml-12"
 				/>
 			</div>
 
 			{/* Title and Link */}
-			<div className="flex-grow border rounded-md p-2">
-				<h2 className="text-lg font-semibold">{title}</h2>
-				<a href={newsLink} className="text-blue-500 hover:underline">
+			<div className="flex-row px-2 items-center mx-10 my-2 md:py-8 xl:flex-col xl:ml-14 xl:mr-8">
+				<h2 className="flex text-base xl:text-3xl font-bold">{title}</h2>
+				<h4 className="flex text-xs xl:text-xl font-semibold">{publishedAt.split('T')[0]}</h4>
+				<a 
+					href = "javascript:void(0);"
+					onClick={(event) => openArticlePage(event)}
+					className="text-blue-500 text-base font-bold xl:text-xl hover:underline"
+				>
 					Read more
 				</a>
 			</div>
 		</div>
 	);
 };
 
+//Line 72-131 are News Fetching
+const fetchNews = async (topic: string, current_date: string) => {

```

Comment:
fetchNews can be moved into NewsPage/api/index.ts

Reply:
Sorry, i feel like this can be moved into src/api/index.ts cus the dashboard may need to use it too